controllers/routemonitor/routemonitor.go

```diff
@@ -63,8 +64,10 @@ func (r *RouteMonitorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	shouldDelete := routeMonitor.WasDeleteRequested()
 	log.V(2).Info("Response of WasDeleteRequested", "shouldDelete", shouldDelete)
 
+	blackboxExporter := blackboxexporter.NewBlackboxExporter(r.Client, log, ctx)
```

Comment:
I think this should be injected from the main and not here

Reply:
done. Although I'd think wrapping Reconcile would be easier as it would reduce the lifetime of objects dramatically.